components/style/properties/longhand/font.mako.rs

```diff
@@ -954,47 +954,104 @@ ${helpers.single_keyword("font-variant-position",
-        use std::fmt;
+        use std::{fmt, str};
         use style_traits::ToCss;
+        use byteorder::{BigEndian, ByteOrder};
+        use cssparser;
 
         impl ToCss for T {
             fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-                match *self {
-                    T::Normal => dest.write_str("normal"),
-                    T::Override(ref lang) => write!(dest, "\"{}\"", lang),
+                if self.0 == 0 {
+                    return dest.write_str("normal")
                 }
+                let mut buf = [0; 4];
+                BigEndian::write_u32(&mut buf, self.0);
+                let slice = if cfg!(debug_assertions) {
+                    str::from_utf8(&buf).unwrap()
+                } else {
+                    unsafe { str::from_utf8_unchecked(&buf) }
+                };
+                cssparser::serialize_string(slice.trim_right(), dest)
             }
         }
 
-        #[derive(Clone, Debug, PartialEq)]
-        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
-        pub enum T {
-            Normal,
-            Override(String),
-        }
+        /// font-language-override can only have a single three-letter
+        /// OpenType "language system" tag, so we should be able to compute
+        /// it and store it as a 32-bit integer
+        /// (see http://www.microsoft.com/typography/otspec/languagetags.htm).
+        pub struct T(pub u32);
     }
 
     #[inline]
     pub fn get_initial_value() -> computed_value::T {
-        computed_value::T::Normal
+        computed_value::T(0)
     }
 
     #[inline]
     pub fn get_initial_specified_value() -> SpecifiedValue {
         SpecifiedValue::Normal
     }
 
+    impl ToComputedValue for SpecifiedValue {
+        type ComputedValue = computed_value::T;
+
+        #[inline]
+        fn to_computed_value(&self, _: &Context) -> computed_value::T {
+            use std::ascii::AsciiExt;
+            match *self {
+                SpecifiedValue::Normal => computed_value::T(0),
+                SpecifiedValue::Override(ref lang) => {
+                    if lang.len() == 0 || lang.len() > 4 || !lang.is_ascii() {
+                        return computed_value::T(0)
+                    }
+                    let mut computed_lang = lang.clone();
+                    while computed_lang.len() < 4 {
+                        computed_lang.push(' ');
+                    }
+                    let bytes = computed_lang.into_bytes();
+                    computed_value::T(BigEndian::read_u32(&bytes))
+                }
+            }
+        }
+        #[inline]
+        fn from_computed_value(computed: &computed_value::T) -> Self {
+            if computed.0 == 0 {
+                return SpecifiedValue::Normal
+            }
+            let mut buf = [0; 4];
+            BigEndian::write_u32(&mut buf, computed.0);
+            SpecifiedValue::Override(String::from_utf8(buf.to_vec()).unwrap())
```

Comment:
This can also use `if cfg!(debug_assertions) { String::from_utf8(..) } else { String::from_utf8_unchecked }` if you want, but no big deal.